src/plugins/launch/slurm/launch_slurm.c

```diff
@@ -622,7 +669,7 @@ static char **_build_user_env(srun_job_t *job, slurm_opt_t *opt_local)
 							    value);
 				}
 			}
-			tok = strtok_r(NULL, ",", &save_ptr);
+			tok = _find_quote_token(NULL, ",", &save_ptr);
```

Comment:
What is wrong with strsep? http://man7.org/linux/man-pages/man3/strsep.3.html Can we please use that instead of our own rolled C code? If you need the save_ptr, you can also do that imho.

Like the man page indicates:

~~~~ 
while((tok = strsep(tmp_env, ",") != NULL) {
    if (xstrcasecmp(tok, "ALL") == 0)
        all = true;
    if (!xstrcasecmp(tok, "NONE"))
        break;
    eq_ptr = strchr(tok, '=');
    if (eq_ptr) {
        eq_ptr[0] = '\0';
        value = eq_ptr + 1;
        env_array_overwrite(&dest_array, tok, value);
    } else {
        value = getenv(tok);
        if (value) {
            env_array_overwrite(&dest_array, tok, value);
        }
    }
}
~~~~

Reply:
this is code copy pasted from exiting slurm code, and schedmd will fix this. why bother reinventing something?